builder/builder_test.go

```diff
@@ -307,3 +339,175 @@ func getAppHashFromEVM(ethState *state.StateDB, header *monomer.Header) (common.
+		InjectedTransactions: txs,
+		GasLimit:             1000000000000,
+		Timestamp:            g.Time + 1,
+		NoTxPool:             true,
+	}
+
+	// Build a block with the payload
+	builtBlock, preBuildInfo, postBuildInfo := buildBlock(t, b, app, payload)
+
+	// Test deposit was received
+	expectedDepositAmount := fmt.Sprintf("%sETH", depositTxETH.Value().String())
+	checkDepositTXResult(t, txStore, depositTxs, expectedDepositAmount, cosmAddr.String())
+
+	withdrawalTxResult, err := txStore.Get(bfttypes.Tx(withdrawalTx).Hash())
+	require.NoError(t, err)
+	require.NotNil(t, withdrawalTxResult)
+	require.Truef(t, withdrawalTxResult.Result.IsOK(), "Expected the withdrawal transaction to be successful, but it failed")
+
+	// Verify block creation
+	genesisBlock, err := blockStore.BlockByHeight(uint64(preBuildInfo.GetLastBlockHeight()))
+	require.NoError(t, err)
+	gotBlock, err := blockStore.HeadBlock()
+	require.NoError(t, err)
+
+	ethStateRoot := gotBlock.Header.StateRoot
+	header := monomer.Header{
+		ChainID:    g.ChainID,
+		Height:     uint64(postBuildInfo.GetLastBlockHeight()),
+		Time:       payload.Timestamp,
+		ParentHash: genesisBlock.Header.Hash,
+		StateRoot:  ethStateRoot,
+		GasLimit:   payload.GasLimit,
+	}
+	wantBlock, err := monomer.MakeBlock(&header, txs)
+	require.NoError(t, err)
+	verifyBlockContent(t, wantBlock, gotBlock, builtBlock)
+
+	// We expect two transactions: one combined transaction (l1InfoTx + depositTxs) and one withdrawal transaction (withdrawalTx).
+	require.Equal(t, 2, builtBlock.Txs.Len(), "Expected the built block to contain 2 transactions: depositTxs and withdrawalTx")
+	require.Equal(t, 2, gotBlock.Txs.Len(), "Expected the built block to contain 2 transactions: depositTxs and withdrawalTx")
+
+	// Test parseWithdrawalMessages
+	testParseWithdrawalMessages(t, withdrawalTxResult)
+
+	expectedStateRoot := wantBlock.Header.StateRoot
+	gotStateRoot := gotBlock.Header.StateRoot
+	require.Equal(t, expectedStateRoot, gotStateRoot, "Expected the built block to contain state root hash")
+
+	for i := range gotBlock.Txs {
+		select {
+		case event, ok := <-subscription.Out():
+			if !ok {
+				require.FailNow(t, "event channel closed unexpectedly")
+			}
+			data := event.Data()
+			require.IsType(t, bfttypes.EventDataTx{}, data)
+			e, ok := data.(bfttypes.EventDataTx)
+			require.True(t, ok)
+			// Deposit transaction
+			if i == 0 {
+				require.Equal(t, e.GetResult().Events[0].Attributes[0].Value, "/rollup.v1.MsgApplyL1Txs")
+			}
+			// Withdrawal transaction
+			if i == 1 {
+				require.Equal(t, e.GetResult().Events[0].Attributes[0].Value, "/rollup.v1.MsgInitiateWithdrawal")
+			}
+
+		case <-subscription.Canceled():
+			require.FailNow(t, "subscription channel closed unexpectedly")
+		}
+		require.NoError(t, subscription.Err())
+	}
+}
+
+func testParseWithdrawalMessages(t *testing.T, withdrawalTxResult *abcitypes.TxResult) {
+	const (
+		nonceKey          = "nonce"
+		expectedEventType = "withdrawal_initiated"
+	)
```

Comment:
Instead of creating new test constants, could you use the actual constants from `x/rollup/types/events.go` for these values?

`AttributeKeyNonce`
`EventTypeWithdrawalInitiated`